src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -2411,15 +2453,19 @@ private static ImmutableArray<Symbol> FixPartialMember(ImmutableArray<Symbol> sy
             SourceMemberMethodSymbol.InitializePartialMethodParts(definition, implementation);
 
             // a partial method is represented in the member list by its definition part:
+            return Remove(symbols, implementation);
+        }
+
+        private static ImmutableArray<Symbol> Remove(ImmutableArray<Symbol> symbols, Symbol symbol)
```

Comment:
Given that replace/original may be used more than partial methods, doing this over and over again can be inefficient (i.e. quadratic). Does it make sense to `RemoveAll` a set of symbols rather than one at a time? (a PROTOTYPE comment or git issue would be a good idea to investigate this later).